File: modules/data/random.lua
sptbl["random"] = {

    files = {
        module = "random.c",
        header = "random.h",
        example = "ex_random.c",
    },

    func = {
        create = "ut_random_create",
        destroy = "ut_random_destroy",
        init = "ut_random_init",
        compute = "ut_random_compute",
    },

    params = {
        optional = {
            {
                name = "min",
                type = "UTFLOAT",
                description = "Minimum value.",
                default = -0.2
            },
            {
                name = "max",
                type = "UTFLOAT",
                description ="Maximum value.",
                default = 0.2
            },
        }
    },

    modtype = "module",

    description = [[Random values within a range]],

    ninputs = 0,
    noutputs = 1,

    inputs = {
        {
            name = "dummmy",
            description = "This does nothing."
        },
    },

    outputs = {
        {
            name = "out",
            description = "Signal output."
        },
    }

}

File: modules/random.c
#include <stdlib.h>
#include "utone.h"

int ut_random_create(ut_random **p)
{
    *p = malloc(sizeof(ut_random));
    return UT_OK;
}

int ut_random_destroy(ut_random **p)
{
    free(*p);
    return UT_OK;
}

int ut_random_init(ut_data *ut, ut_random *p)
{
    p->min = -1;
    p->max = 1;
    return UT_OK;
}

int ut_random_compute(ut_data *ut, ut_random *p, UTFLOAT *in, UTFLOAT *out)
{
    /* Send the signal's input to the output */
    UTFLOAT rnd = ((ut_rand(ut) % RAND_MAX) / (RAND_MAX * 1.0));
    rnd *= (p->max - p->min);
    rnd += p->min;
    *out = rnd;
    return UT_OK;
}
